### ISSUE TITLE
update docs to show how to use Travis-CI's new container-based jobs

### ISSUE THREAD

**Issue by hvr** [hvr/multi-ghc-travis#17]
- http://blog.travis-ci.com/2015-03-31-docker-default-on-the-way/
- http://docs.travis-ci.com/user/apt/

**Comment by hvr** [comment]
This depends on travis-ci/travis-ci#3560

**Comment by FranklinChen** [comment]
Would love to see this happen soon!

**Comment by emwap** [comment]
travis-ci/travis-ci#3560 is fixed. I have made a trial in one of my repos, https://github.com/emwap/plugins-multistage/blob/master/.travis.yml

However, it seems that the environment variables are not available when defining apt dependencies, so I had to install all compilers needed for the entire matrix.

**Comment by hvr** [comment]
@emwap you may want to take a look at travis-ci/travis-ci#3505

**Comment by emwap** [comment]
Nice! A pity with the repetition of version numbers though. It's easy to mess that up.
I tried it successfully on my repo above,

**Comment by hvr** [comment]
Downloading the ppa packages seems to be surprisingly slow though (at roughly ~100KiB/s) :-/

**Comment by emwap** [comment]
Yes, I don't know if that is erratic behavior of their caching. I have seen figures in that range, but also around 3 MiB/s.

**Comment by juanpaucar** [comment]
Hi guys, i was wondering, since now that any build on travis tells us about being run on legacy infrastructure; maybe it's a good time to start considering making a PR for these changes. :smile:

**Comment by treeowl** [comment]
Well, now it seems they've changed their minds and shut down the container-based infrastructure. Maybe time to update the docs (and code?) to *remove* mention of it?
